packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -313,12 +333,63 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
 
   private updateChatAdapter(chatAdapter: ChatAdapter): void {
     this.chatAdapter?.offStateChange(this.onChatStateChange);
+    this.updateChatEventListeners(chatAdapter);
     this.chatAdapter = chatAdapter;
     this.chatAdapter.onStateChange(this.onChatStateChange);
     this.context.updateClientStateWithChatState(chatAdapter.getState());
     this.emitter.emit('chatInitialized', this.chatAdapter);
   }
 
+  private updateChatEventListeners(chatAdapter: ChatAdapter): void {
+    const messageReceivedListeners = this.getListeners('messageReceived');
```

Comment:
if someone adds a new event it will be easy to miss adding it here, can you instead more blindly paste the listeners over?
e.g. (untested)
```
for (const [eventName, listeners] of this.chatEventListeners) {
  for (const listener of listeners) {
    this.chatAdapter?.off('eventName', listener);
    chatAdapter.on('eventName', listener);
  }
}
```
(might have to use `as any` to satisfy typescript